server/game/gameSystems/DiscardFromDeckSystem.ts

```diff
@@ -0,0 +1,89 @@
+import { AbilityContext } from '../core/ability/AbilityContext';
+import { Card } from '../core/card/Card';
+import { EventName, GameStateChangeRequired } from '../core/Constants';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem';
+import Player from '../core/Player';
+import { DiscardSpecificCardSystem } from './DiscardSpecificCardSystem';
+import * as Contract from '../core/utils/Contract';
+
+export interface IDiscardFromDeckProperties extends IPlayerTargetSystemProperties {
+    amount?: number;
+}
+
+export class DiscardFromDeckSystem<TContext extends AbilityContext = AbilityContext> extends PlayerTargetSystem<TContext, IDiscardFromDeckProperties> {
```

Comment:
Do you think `DiscardFromTopOfDeck` would be worth it for the added clarity?

Reply:
@AMMayberry1 AFAIK there isn't currently any way to discard from anywhere else in the deck, so I think it's fine for now. If we ever have other discard from deck options, we can add some sort of a DiscardMode to this. Does that sound good?

Reply:
Yup makes sense